File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
#include<unordered_set>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        if(nums.size()==0){
            return 0;
        }
        unordered_set<int>s;
        for(int i=0;i<nums.size();i++){
            if(s.count(nums[i])){
                return 1;
            }
            s.insert(nums[i]);
        }
        return 0;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums)==0:
            return 0
        #use unordered set
        hash_set=set(nums)#put all elements
        ans=1
        for start in nums:
            #is start even possible?
            if (start-1) in hash_set:
                continue #nope
            count=1
            start+=1
            while start in hash_set:
                count+=1
                start+=1
            ans=max(ans,count)
        return ans
            
                   



        
        